modules/product/src/main/java/com/opengamma/strata/product/bond/ResolvedFixedCouponBondOption.java

```diff
@@ -0,0 +1,550 @@
+/*
+ * Copyright (C) 2022 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.bond;
+
+import java.io.Serializable;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.joda.beans.Bean;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.gen.ImmutableValidator;
+import org.joda.beans.gen.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.ResolvedProduct;
+import com.opengamma.strata.product.common.LongShort;
+
+/**
+ * An option on a {@link FixedCouponBond} resolved for pricing.
+ * <p>
+ * The option strike is expressed as "Clean price". The "clean" price excludes any accrued interest.
+ * <p>
+ * The call/put is provided by the quantity's sign. If positive, it indicates the right to buy the bond (call), 
+ * if negative it indicates the right to sell the bond (put).
+ */
+@BeanDefinition
+public final class ResolvedFixedCouponBondOption
+    implements ResolvedProduct, ImmutableBean, Serializable {
+
+  /**
+   * Whether the option is long or short.
+   * <p>
+   * Long indicates that the owner has the right to exercise the option at expiry.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LongShort longShort;
+  /**
+   * The bond underlying the option.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ResolvedFixedCouponBond underlying;
+  /**
+   * The expiry date-time of the option.
+   * <p>
+   * The option is European, and can only be exercised on the expiry date.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ZonedDateTime expiry;
+  /**
+   * The quantity that was traded.
+   * <p>
+   * This will be positive if buying (call) and negative if selling (put).
+   */
+  @PropertyDefinition
+  private final double quantity;
```

Comment:
do we need quantity on both the underlying and the resolved version. we can add a getter on resolved to access the underlying quantity

Reply:
There is a "notional" on the FixedCouponBond, but this is intended for "tradable size". It most cases it is simply 1, but for some bond, they can be traded only in multiples of 1,000 or 1,000,000. The FixedCouponBondTrade have also a similar "quantity".